lib/api/objectPutRetention.js

```diff
@@ -80,7 +82,51 @@ function objectPutRetention(authInfo, request, log, callback) {
                 (err, retentionInfo) => next(err, bucket, retentionInfo, objectMD));
         },
         (bucket, retentionInfo, objectMD, next) => {
-            const validationError = validateObjectLockUpdate(objectMD, retentionInfo);
+            if (objectMD.retentionMode === 'GOVERNANCE' && authInfo.isRequesterAnIAMUser()) {
+                log.trace('object in GOVERNANCE mode and is user, checking for attached policies',
+                    { method: 'objectPutRetention' });
+                const authParams = auth.server.extractParams(request, log, 's3',
+                    request.query);
+                const ip = policies.requestUtils.getClientIp(request, config);
+                const requestContextParams = {
+                    constantParams: {
+                        headers: request.headers,
+                        query: request.query,
+                        generalResource: bucketName,
+                        specificResource: { key: objectKey },
+                        requesterIp: ip,
+                        sslEnabled: request.connection.encrypted,
+                        apiMethod: 'bypassGovernanceRetention',
+                        awsService: 's3',
+                        locationConstraint: bucket.getLocationConstraint(),
+                        requesterInfo: authInfo,
+                        signatureVersion: authParams.params.data.authType,
+                        authType: authParams.params.data.signatureVersion,
```

Comment:
Swap here

Reply:
makes me think that some specific tests to catch this particular typo issue could be useful (not sure what to test exactly, but I'm surprised that Vault would return a success if we mess with those input parameters)

Reply:
After some testing, the value of these params doesn't matter as the values for aren't used (though they still need to be present or vault will reject the request with `ValidationError`). The same error also exists in bucketPut and multiObjectDelete.